sdks/python/apache_beam/runners/worker/log_handler.py

```diff
@@ -77,14 +83,14 @@ def emit(self, record):
     nanoseconds = 1e9 * fraction
     log_entry.timestamp.seconds = int(seconds)
     log_entry.timestamp.nanos = int(nanoseconds)
-    self._log_entry_queue.put(log_entry)
+    self._log_entry_queue.put(log_entry, block=False)
 
   def close(self):
     """Flush out all existing log entries and unregister this handler."""
     # Acquiring the handler lock ensures ``emit`` is not run until the lock is
     # released.
     self.acquire()
-    self._log_entry_queue.put(self._FINISHED)
+    self._log_entry_queue.put(self._FINISHED, timeout=1)
```

Comment:
I think we need this to be blocking, to ensure that existing buffers is fully flushed out.

Reply:
The default is blocking, so we'll block for up to 1 second. The issue I see is that we probably don't want to indefinitely block shutdown in the case where we're unable to re-connect to the logging service.

Reply:
Sounds reasonable.